sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/RequestResponseChannel.java

```diff
@@ -200,8 +200,8 @@ protected RequestResponseChannel(AmqpConnection amqpConnection, String connectio
 
     @Override
     public void dispose() {
-        disposeAsync("Dispose called.")
-            .block(retryOptions.getTryTimeout());
```

Comment:
This is to fix an exception
```
class com.azure.core.amqp.implementation.RequestResponseChannel:$cbs - Error occurred closing Disposable channel.
block()/blockFirst()/blockLast() are blocking, which is not supported in thread reactor-executor-1
java.lang.IllegalStateException: block()/blockFirst()/blockLast() are blocking, which is not supported in thread reactor-executor-1
	at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:117)
	at reactor.core.publisher.Mono.block(Mono.java:1727)
	at com.azure.core.amqp.implementation.RequestResponseChannel.dispose(RequestResponseChannel.java:204)
	at com.azure.core.amqp.implementation.AmqpChannelProcessor.close(AmqpChannelProcessor.java:309)
	at com.azure.core.amqp.implementation.AmqpChannelProcessor.setAndClearChannel(AmqpChannelProcessor.java:297)
	at com.azure.core.amqp.implementation.AmqpChannelProcessor.dispose(AmqpChannelProcessor.java:256)
	at com.azure.core.amqp.implementation.ReactorConnection.dispose(ReactorConnection.java:363)
```

Reply:
* This will never be subscribed to. `disposeAsync()` does nothing without a subscription. 
* This method would end prematurely because it would hook up and kick off the operation and then not wait for all the resources to actually be disposed of.

Reply:
We should be able to change this to `disposeAsync("Dispose called.").subscribe()`, in order to kick off this operation.

Reply:
When `dispose()` is called it has to subscribe to `disposeAsync()` and `block` until it's complete. However, the error that @YijunXieMS  is seeing seems like the `dispose` method is being called from non-blocking reactor thread. So, the caller should then use `disposeAsync()` method directly instead of calling `dispose`.

Reply:
Yeah. I was wondering if we needed to implement AutoCloseable at all in this case since dispose() isn't really the right option.